HLTriggerOffline/Muon/src/HLTMuonValidator.cc

```diff
@@ -63,86 +51,92 @@ class HLTMuonValidator : public DQMEDAnalyzer {
   HLTMuonPlotter::L1MuonMatcherAlgoForDQM const l1tMuonMatcherAlgo_;
 };
 
-//////////////////////////////////////////////////////////////////////////////
-//////// Namespaces, Typedefs, and Constants /////////////////////////////////
+HLTMuonValidator::HLTMuonValidator(const edm::ParameterSet &pset)
+    : pset_(pset),
+      hltProcessName_(pset.getParameter<std::string>("hltProcessName")),
+      hltPathsToCheck_(pset.getParameter<std::vector<std::string>>("hltPathsToCheck")),
+      triggerEventToken_(consumes(edm::InputTag("hltTriggerSummaryRAW"))),
+      genParticlesToken_(consumes(pset.getParameter<std::string>("genParticleLabel"))),
+      recoMuonsToken_(consumes(pset.getParameter<std::string>("recMuonLabel"))),
+      l1tMuonMatcherAlgo_(pset, consumesCollector()) {}
 
-using namespace std;
-using namespace edm;
-using namespace reco;
-using namespace trigger;
+void HLTMuonValidator::fillLabels(std::string const &path,
+                                  std::vector<std::string> &moduleLabels,
+                                  std::vector<std::string> &stepLabels) const {
+  auto const &hltFilters = hltConfig_.saveTagsModules(path);
 
-typedef vector<string> vstring;
+  moduleLabels.clear();
+  moduleLabels.reserve(hltFilters.size());
 
-//////////////////////////////////////////////////////////////////////////////
-//////// Class Methods ///////////////////////////////////////////////////////
+  stepLabels.clear();
+  stepLabels.reserve(hltFilters.size() + 1);
 
-HLTMuonValidator::HLTMuonValidator(const ParameterSet &pset)
-    : pset_(pset),
-      hltProcessName_(pset.getParameter<string>("hltProcessName")),
-      hltPathsToCheck_(pset.getParameter<vstring>("hltPathsToCheck")),
-      triggerEventToken_(consumes(edm::InputTag("hltTriggerSummaryRAW"))),
-      genParticlesToken_(consumes(pset.getParameter<string>("genParticleLabel"))),
-      recoMuonsToken_(consumes(pset.getParameter<string>("recMuonLabel"))),
-      l1tMuonMatcherAlgo_(pset, consumesCollector()) {}
+  for (auto const &module : hltFilters) {
+    if (module.find("Filtered") == std::string::npos)
+      continue;
 
-vector<string> HLTMuonValidator::moduleLabels(string path) {
-  vector<string> modules = hltConfig_.moduleLabels(path);
-  vector<string>::iterator iter = modules.begin();
+    auto const step_label = stepLabel(module);
+    if (step_label.empty() or std::find(stepLabels.begin(), stepLabels.end(), step_label) != stepLabels.end())
```

Comment:
This requires that there are no duplicates in the `stepLabels` vector. Afaiu, having duplicates could cause histograms of a certain type, e.g. `_L3`, to be filled more often than histograms of type `_All` (no matching requirements), and this could spoil the efficiency plots (numerator with more entries than denominator).